test/unit/workers/document_list_export_worker_test.rb

```diff
@@ -52,4 +52,24 @@ class DocumentListExportWorkerTest < ActiveSupport::TestCase
     MailNotifications.expects(:document_list).with(regexp_matches(%r{^https://whitehall-admin\.test\.gov\.uk/export/documents/[a-f0-9-]{36}$}), @user.email, title).returns(stub(deliver_now: nil))
     @worker.perform({ "state" => "draft" }, @user.id)
   end
+
+  test "does not raise an error if an email cannot be sent via notify" do
+    raises_exception = lambda { |_url, _address, _title|
+      response = MiniTest::Mock.new
+      ENV["SENTRY_CURRENT_ENV"] = "integration"
+      response.expect :code, 400
+      response.expect :body, "Can't send to this recipient using a team-only API key"
+      raise Notifications::Client::BadRequestError, response
+    }
```

Comment:
This test took me a few reads 🤔  so it looks like you're stubbing that `MailNotifications.document_list` raises an exception, and you're asserting that the status code and body associated with the response are what you expect.

But it looks like we're missing an assertion that the error doesn't get re-raised... unless I'm missing something?

For comparison, we should also have a test where we set `SENTRY_CURRENT_ENV` to "production".

Reply:
I had to copy it from the previous test I wrote (in test/functional/admin/fact_check_requests_controller_test.rb) because I no longer remembered how testing an exception in minitest works.

I _think_ that because the test finishes at all, it indicates that the error is not raised (or is caught and handled) — changing `SENTRY_CURRENT_ENV` causes the test to fail, for example. But it could probably be written in a way that explicitly asserts that no error is raised (I seem to recall a discussion about that recently).

I'll add another with the env as "production" where it asserts that an exception _is_ raised.